app/src/main/java/com/example/songme/ui/adapter/TrackAdapter.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.songme.ui.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.songme.R
+import com.example.songme.data.model.Track
+import kotlinx.android.synthetic.main.item_track.view.*
+
+class TrackAdapter : RecyclerView.Adapter<TrackAdapter.ViewHolder>() {
+
+    var onItemClick: (Track, Int) -> Unit = { _, _ -> }
+    private val tracks = mutableListOf<Track>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
+        ViewHolder(
+            LayoutInflater.from(parent.context).inflate(R.layout.item_track, parent, false),
+            onItemClick
+        )
+
+    override fun getItemCount(): Int = tracks.size
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bindData(tracks[position])
+    }
+
+    fun updateData(newTracks: List<Track>) {
+        tracks.clear()
+        tracks.addAll(newTracks)
+        notifyDataSetChanged()
+    }
+
+    class ViewHolder(itemView: View, onItemClick: (Track, Int) -> Unit) :
+        RecyclerView.ViewHolder(itemView) {
+
+        private var itemTrack: Track? = null
+
+        init {
+            itemView.setOnClickListener { itemTrack?.let { onItemClick(it, adapterPosition) } }
+        }
+
+        fun bindData(track: Track) {
+            itemTrack = track
+            itemView.run {
+                Glide.with(itemView)
+                    .load(track.imageUrl)
+                    .error(R.drawable.ic_music_black)
+                    .into(imageTrack)
+                titleTrack.text = track.title
+                titleSinger.text = track.author
+            }
```

Comment:
```suggestion
                setOnClickListener { onItemClick(track, adapterPosition) }
            }
```

Reply:
e đã thử sửa rồi, nhưng e thấy onItemClick hoàn toàn k nằm trên view giống như titleTrack, titleSinger. nó là một tham số e truyền vào ViewHolder nên e không thể gọi đến nó trong bindData
![error](https://user-images.githubusercontent.com/51770796/81395521-fc529000-914d-11ea-9ed4-f28cd082721a.PNG)

Reply:
Lỗi này là do ông bỏ `private val `trên `onItemClick` trong `constructor`

Reply:
không setOnCLick trong onBindData, set trong init

Reply:
t set trong onBindData đc rồi, nãy t để trong init nhưng các a comment nên t mới chuyển vào bindData